src/frontend/sharedComponents/MediaScrollView/PhotoThumbnail.tsx

```diff
@@ -13,38 +12,62 @@ import {
 import {AlertIcon} from '../icons';
 import debug from 'debug';
 import {LIGHT_GREY} from '../../lib/styles';
+import {
+  DraftPhoto,
+  Photo,
+  SavedPhoto,
+} from '../../contexts/PhotoPromiseContext/types';
+import {useAttachmentUrlQuery} from '../../hooks/server/media';
 
 const log = debug('Thumbnail');
 
-type PhotoThumbnailProps = {
-  photo?: Partial<Photo>;
-  onPress: () => any;
-  style?: StyleProp<ViewStyle>;
-  size?: number;
+type PhotoThumbnailProps = PhotoThumbnailContextProps & {
+  photo: Photo;
 };
 
 export const PhotoThumbnail: FC<PhotoThumbnailProps> = props => {
   const {photo, style, size, onPress} = props;
-  const [error, setError] = React.useState(false);
 
-  const uri =
-    (photo && 'thumbnailUri' in photo && photo.thumbnailUri) || undefined;
-  const photoProcessingError = photo && 'error' in photo && !!photo.error;
-  const isCapturing = photo && photo.type === 'unprocessed';
+  return (
+    <PhotoThumbnailContext.Provider value={{style, size, onPress}}>
+      {photo.type === 'photo' ? (
+        <SavedPhotoThumbnail photo={photo} />
+      ) : (
+        <UnsavedPhotoThumbnail photo={photo} />
+      )}
+    </PhotoThumbnailContext.Provider>
```

Comment:
nit: personally, would lean towards not using context for this and instead either:

1. settle for a small amount of prop drilling
2. update the internal components so that they're easier to use without passing props excessively (probably more refactor-like work, similar to what i did in https://github.com/digidem/comapeo-mobile/pull/491)

Reply:
Also, I agree that contexts are a little unnecessary when only prop drilling 2 levels. And i actually originally designed it to use prop drilling. But doing a comparison of prop drilling and context, in this case, I actually think the context simplifies things. Because of the `useAttachmentUrlQuery` hook, we have to break each thumbnail into its own component. Both have the same shared prop. prop drilling is doubled since there are 2 component, even though the are the exact same thing. Since we are drillling it 2 levels, they are passed 4 times. It made the code feel unnecessarily verbose.